cmd/provisioner-localpv/app/helper.go

```diff
@@ -81,13 +87,90 @@ func (p *Provisioner) getPathAndNodeForPV(pv *v1.PersistentVolume) (path, node s
 	return path, node, nil
 }
 
+// createInitPod launches a helper(busybox) pod, to create the host path.
+//  The local pv expect the hostpath to be already present before mounting
+//  into pod. Validate that the local pv host path is not created under root.
+func (p *Provisioner) createInitPod(pOpts *HelperPodOptions) (err error) {
+	if pOpts.name == "" || pOpts.path == "" || pOpts.nodeName == "" {
+		return fmt.Errorf("invalid empty name or path or node")
+	}
+
+	// Initialize HostPath builder and validate that
+	// non-root directories are not passed for create
+	// Extract the base path and the volume unique path.
+	parentDir, volumeDir, vErr := hostpath.NewBuilder().WithPath(pOpts.path).
+		WithCheckf(hostpath.IsNonRoot(), "path should not be a root directory: path %v", pOpts.path).
+		ExtractSubPath()
+	if vErr != nil {
+		return vErr
+	}
+
+	conObj, _ := mContainer.Builder().
+		WithName("local-path-init").
+		WithImage(p.helperImage).
+		WithCommand(append(pOpts.cmdsForPath, filepath.Join("/data/", volumeDir))).
```

Comment:
Need to think how this can be more intuitive. From this calling code, I am not able to figure out whats happening? Hows the command is getting built.